vyper/contracts/TokenTransferer.vy

```diff
@@ -0,0 +1,75 @@
+# @version 0.3.3
+
+struct ConduitBatch1155Transfer:
+    token: address
+    _from: address
+    to: address
+    ids: DynArray[uint256, 10] 
+    amounts: DynArray[uint256, 10] 
+
+TRANSFERFROM_MID: constant(Bytes[4]) = method_id("transferFrom(address,address,uint256)")
+TRANSFERFROM1155_MID: constant(Bytes[4]) = method_id("transferFrom(address,address,uint256,uint256)")
+SAFEBATCHTRANSFERFROM_MID: constant(Bytes[4]) = method_id("safeBatchTransferFrom(address,address,uint256[],uint256[],bytes)")
+
+@internal
+def _performERC20Transfer(token: address, _from: address, to: address, amount: uint256):
+    _response: Bytes[32] = raw_call(
+    token,
+    _abi_encode(
+        TRANSFERFROM_MID,
+        _from,
+        to,
+        amount
+        ),
+        max_outsize=32
+    )  
+    if len(_response) > 0:
+        assert convert(_response, bool), "TransferFrom failed"  
+
+@internal
+def _performERC721Transfer(token: address, _from: address, to: address, identifier: uint256):
+    _response: Bytes[32] = raw_call(
+    token,
+    _abi_encode(
```

Comment:
discussed offline: we should go with using `IERC721(token).transfeFrom()` instead of `raw_call`.

Reply:
I think so, that way codesize check will be performed and it'll be more readable